tests/ffi/tests.cc

```diff
@@ -203,13 +203,37 @@ void c_take_rust_vec_shared(rust::Vec<Shared> v) {
   }
 }
 
+void c_take_rust_vec_shared_forward_iterator(rust::Vec<Shared> v) {
+  // Exercise requirements of ForwardIterator
+  // https://en.cppreference.com/w/cpp/named_req/ForwardIterator
+  uint32_t sum = 0;
+  for (auto it = v.begin(); it != v.end(); it++) {
```

Comment:
The ForwardIterator behavior is already covered by the previous test in this file which desugars to:

```cpp
for (auto __begin = v.begin(), __end = v.end(); __begin != __end; ++__begin) {
  /* ... */
}
```

I don't think it's worthwhile to keep an additional test like this, but if you think this is something people would write, we should make it call v.end() in the initializer so that anyone copying the code will avoid an extra 3 foreign calls on every iteration.

Reply:
This tests two things on the `const_iterator` that the other test doesn't cover: the postfix ++ operator and the the arrow operator.  

The arrow operator might not be a requirement for the forward iterator.  You mentioned it in your example so I included it:

```c++
void c_take_rust_vec_shared(rust::Vec<Shared> v) {
  auto it = v.begin();
  (void)it++;
  (void)it->z;
```

I went ahead and made the test more efficient in 7366d7f.